scipy/sparse/compressed.py

```diff
@@ -614,14 +614,32 @@ def __setitem__(self, index, x):
         i, j = self._index_to_arrays(i, j)
 
         if isspmatrix(x):
-            x = x.toarray()
-
-        # Make x and i into the same shape
-        x = np.asarray(x, dtype=self.dtype)
-        x, _ = np.broadcast_arrays(x, i)
+            broadcast_row = x.shape[0] == 1 and i.shape[0] != 1
```

Comment:
Hmm, probably can't happen due to `_index_to_arrays` (I don't remember any more exactly how all this worked :)